File: api/Models/Response/TeamSelectionData.cs
using SpoRE.Infrastructure.Database;

namespace SpoRE.Models.Response;

public record TeamSelectionData(
    int Budget,
    int BudgetOver,
    DateTime RaceStart,
    IEnumerable<RiderParticipation> Team,
    IEnumerable<SelectableRider> AllRiders,
    IEnumerable<string> AllTeams);

File: client/src/pages/teamselection/Models/TeamSelectionData.ts
import type { RiderParticipation } from "../../../models/RiderParticipation";
import type { SelectableRider } from "./SelectableRider";

export interface TeamSelectionData {
  budget: number;
  budgetOver: number;
  team: RiderParticipation[];
  allRiders: SelectableRider[];
  allTeams: string[];
}
